network/src/protocols/health_checker/mod.rs

```diff
@@ -138,10 +141,8 @@ pub struct HealthChecker {
     network_context: Arc<NetworkContext>,
     /// A handle to a time service for easily mocking time-related operations.
     time_service: TimeService,
-    /// Channel to send requests to Network layer.
-    network_tx: HealthCheckerNetworkSender,
-    /// Channel to receive notifications from Network layer about new/lost connections.
-    network_rx: HealthCheckerNetworkEvents,
+    /// Network interface to send requests to the Network Layer
+    network_interface: HealthCheckNetworkInterface,
     /// Map from connected peer to last round of successful ping, and number of failures since
     /// then.
     connected: HashMap<PeerId, (u64, u64)>,
```

Comment:
It kinda feels like maybe we should make the PeerDB support a generic interface that contains Peer to PeerAppData so that we can merge this into that space as well -- otherwise we're still going to be maintaining this state everywhere and then we're trading a tens of lines of code for hundreds.

Reply:
e.g., PeerDB<T> and support peer_data_read(&self, peer_id) -> &T and peer_data_write(&self, peer_id, value: T)

then we can specialize it in the local space

Reply:
I got a little flak for centralizing this kind of data, but let me see what I can do.